pkg/mimir/mimir.go

```diff
@@ -818,19 +832,18 @@ func (t *Mimir) Run() error {
 	return err
 }
 
-func (t *Mimir) readyHandler(sm *services.Manager) http.HandlerFunc {
+func (t *Mimir) readyHandler(sm *services.Manager, ready *atomic.Bool) http.HandlerFunc {
```

Comment:
Following up Peter's comment, I would rename it here too (and invert the logic). This will allow to provide a better  response message and log (see below).

```suggestion
func (t *Mimir) readyHandler(sm *services.Manager, shutdownRequested *atomic.Bool) http.HandlerFunc {
```